ringbuf/reader.go

```diff
@@ -0,0 +1,255 @@
+package ringbuf
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"runtime"
+	"sync"
+
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/internal"
+	"github.com/cilium/ebpf/internal/unix"
+)
+
+var (
+	errClosed  = errors.New("ringbuf reader was closed")
```

Comment:
I think this should be exported, so that users can detect `reader.Close` via `errors.Is(err, ringbuf.ErrClosed)`

Reply:
changed. Added `errors.Is` in the tests too.